src/lib.rs

```diff
@@ -53,7 +54,26 @@ If you are looking for a more high-level solution - checkout
 #![allow(ellipsis_inclusive_range_patterns)]
 
 
-use std::fmt;
+/// A facade around all the types we need from the `std`, `core`, and `alloc`
```

Comment:
I think we should use `core` instead of `std` everywhere. And 

```rust
#[cfg(feature = "std")]
extern crate std;
```

in the `lib.rs`, plus:

```rust
#[cfg(feature = "std")]
use std::string::String; // And others.
```

when needed.